openeo/extra/stac_job_db.py

```diff
@@ -0,0 +1,303 @@
+import concurrent
+import datetime
+import logging
+from typing import Iterable, List, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import pystac
+import requests
+from pystac import Collection, Item
+from pystac_client import Client
+from requests.auth import HTTPBasicAuth
+from shapely.geometry import mapping, shape
+
+from openeo.extra.job_management import JobDatabaseInterface, MultiBackendJobManager
+
+_log = logging.getLogger(__name__)
+
+
+class STACAPIJobDatabase(JobDatabaseInterface):
+    """
+    Persist/load job metadata from a STAC API
+
+    Unstable API, subject to change.
+
+    :implements: :py:class:`JobDatabaseInterface`
+    """
+
+    def __init__(
+        self,
+        collection_id: str,
+        stac_root_url: str,
+        auth: requests.auth.AuthBase,
+        has_geometry: bool = False,
+        geometry_column: str = "geometry",
+    ):
+        """
+        Initialize the STACAPIJobDatabase.
+
+        :param collection_id: The ID of the STAC collection.
+        :param stac_root_url: The root URL of the STAC API.
+        :param auth: requests AuthBase that will be used to authenticate, e.g. OAuth2ResourceOwnerPasswordCredentials
+        :param has_geometry: Whether the job metadata supports any geometry that implements __geo_interface__.
+        :param geometry_column: The name of the geometry column in the job metadata that implements __geo_interface__.
+        """
+        self.collection_id = collection_id
+        self.client = Client.open(stac_root_url)
+
+        self._auth = auth
+        self.has_geometry = has_geometry
+        self.geometry_column = geometry_column
+        self.base_url = stac_root_url
+        self.bulk_size = 500
+
+
+
+    def exists(self) -> bool:
+        return len([c.id for c in self.client.get_collections() if c.id == self.collection_id]) > 0
```

Comment:
FYI: this is a bit simpler and cheaper:
```suggestion
        return any(c.id == self.collection_id for c in self.client.get_collections())
```